File: backend/src/constants/subscriptionConstants.js
/** RevenueCat ürün kimlikleri — App Store Connect / Play Console ile eşleşmeli. */
export const SUBSCRIPTION_PRODUCT_IDS = {
  monthly: "aidatpanel_monthly",
  annual: "aidatpanel_annual",
  businessMonthly: "aidatpanel_business_monthly",
  businessAnnual: "aidatpanel_business_annual",
};

/** DB / webhook plan değerleri */
export const SUBSCRIPTION_PLANS = [
  "monthly",
  "annual",
  "business_monthly",
  "business_annual",
];

export const SUBSCRIPTION_PLATFORMS = ["ios", "android"];

/** Temel plan (monthly / annual) bina üst sınırı */
export const BUILDING_LIMIT_BASIC = 20;

/** Business plan — null = sınırsız */
export const BUILDING_LIMIT_BUSINESS = null;

/** Abonelik yok / süresi dolmuş — yeni bina eklenemez */
export const BUILDING_LIMIT_NONE = 0;

export function isBusinessPlan(plan) {
  return plan === "business_monthly" || plan === "business_annual";
}

export function isBasicPlan(plan) {
  return plan === "monthly" || plan === "annual";
}

export function isEntitledStatus(status) {
  return status === "ACTIVE" || status === "TRIAL";
}

File: mobile/lib/core/constants/subscription_constants.dart
/// RevenueCat / mağaza ürün kimlikleri — backend ile eşleşmeli.
abstract final class SubscriptionConstants {
  static const String monthlyProductId = 'aidatpanel_monthly';
  static const String annualProductId = 'aidatpanel_annual';
  static const String businessMonthlyProductId = 'aidatpanel_business_monthly';
  static const String businessAnnualProductId = 'aidatpanel_business_annual';

  static const int basicBuildingLimit = 20;

  static const List<String> allProductIds = [
    monthlyProductId,
    annualProductId,
    businessMonthlyProductId,
    businessAnnualProductId,
  ];
}
